### PR TITLE
Catch version selector

### PR DESCRIPTION
**Pull request by dpshelio**
as said above. Problem was found at astro-informatics/purify#190 as in astro-informatics/purify#183 was failing after an update on Catch2.

### CODE REVIEW

**Comment by dpshelio** [comment]
It needs to update the [wiki page](https://github.com/UCL/GreatCMakeCookOff/wiki/Adding-the-Catch-testing-framework) too.

File: modules/LookUpCatch.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+if(NOT Catch_URL)
+  set(Catch_URL https://raw.githubusercontent.com/catchorg/Catch2/${Catch_URL_VERSION}/single_include/catch2/catch.hpp)
+endif()
+
+set(Catch_FILE "${EXTERNAL_ROOT}/include/catch.hpp")
+file(MAKE_DIRECTORY "${EXTERNAL_ROOT}/include")
+file(DOWNLOAD ${Catch_URL} "${Catch_FILE}")
+
+file(READ "${Catch_FILE}" CATCHSTRING LIMIT 1000)
```
Comment (jonc125) [inline modules/LookUpCatch.cmake]: You should add a comment explaining what this next section is doing (checking download worked by ensuring the file is long enough).

File: scripts/AddCatchTest.cmake [hunk 2]
```diff
@@ -1,33 +1,9 @@
...
-    string(LENGTH "${CATCHSTRING}" CATCHLENGTH)
-    if(NOT CATCHLENGTH GREATER 500)
-      file(REMOVE "${catch_file}")
-      message(FATAL_ERROR "Failed to download Catch ${CATCHSTRING} ${CATCHLENGTH}")
-    endif()
-  endif()
-  find_package(Catch REQUIRED)
+if(NOT Catch_FOUND AND Catch_WANTED_VERSION)
+  lookup_package(Catch REQUIRED ARGUMENTS VERSION ${Catch_WANTED_VERSION})
+else(NOT Catch_FOUND)
```
Comment (jonc125) [inline scripts/AddCatchTest.cmake]: Shouldn't this be `elseif`?

File: modules/LookUpCatch.cmake (hunk 1 above)
Comment (dpshelio) [reply modules/LookUpCatch.cmake]: Yeah... that was on the original file. I imagine is checking whether the file is long enough as I suppose redirects or errors will be shorter ones.

File: scripts/AddCatchTest.cmake (hunk 2 above)
Comment (dpshelio) [reply scripts/AddCatchTest.cmake]: I think you are right... though I don't understand what the expression on `else` is for. Maybe just to know in which `if` block you are in.

**Comment by dpshelio** [comment]
Updated comments by @jonc125  - Thanks!!

**Pull request merged**